app/questionnaire/questionnaire_schema.py

```diff
@@ -908,7 +908,7 @@ def _populate_when_rules_section_dependencies(self) -> None:
                     ):
                         self._when_rules_section_dependencies_by_section_for_progress_value_source[
                             key
-                        ].append(
+                        ].add(
```

Comment:
Whilst it's always one element, using add(*ids) looks risky if anything were to change - should this be using update instead?

Reply:
Updated 👍